Content/Items/Magnet/Weapons.ThunderBeads.cs

```diff
@@ -0,0 +1,280 @@
+﻿using log4net.Core;
+using ReLogic.Content;
+using StarlightRiver.Helpers;
+using System;
+using System.Collections.Generic;
+using Terraria.Audio;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
+using Terraria.Graphics.Effects;
+using static System.Windows.Forms.VisualStyles.VisualStyleElement.TaskbarClock;
+using StarlightRiver.Content.Dusts;
+using System.Runtime.InteropServices;
+
+namespace StarlightRiver.Content.Items.Magnet
+{
+	public class ThunderBeads : ModItem
+	{
+		public override string Texture => AssetDirectory.MagnetItem + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Magnet Beads");
+			Tooltip.SetDefault("Whip enemies to stick the beads in them \nRepeatedly click to shock affected enemies");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = 1;
+		}
+
+		public override void SetDefaults()
+		{
+			Item.DefaultToWhip(ModContent.ProjectileType<ThunderBeads_Whip>(), 15, 1.2f, 5f, 25);
+			Item.value = Item.sellPrice(0, 2, 0, 0);
+			Item.rare = ItemRarityID.Orange;
+		}
+	}
+
+	public class ThunderBeads_Whip : BaseWhip
```

Comment:
The underscore here violates naming convention, this should likely be changed to ThunderBeadsProjectile or something similar

Reply:
Fixed